src/cli/java/org/commcare/util/cli/ApplicationHost.java

```diff
@@ -258,6 +259,9 @@ private boolean loopSession() throws IOException {
 
                     //Restart
                     return true;
+                } catch (InvalidCaseGraphException e) {

```

Comment:
This doesn't quite seem like the right place to catch this to me. The other two exceptions are broad-context exceptions that could get thrown in any number of circumstances, but this one is highly specific to one or two states of execution

Reply:
I think it's fine to start with just "Catch and Rethrow as a Runtime Exception" for now, or to simply print the error message and move on (as is done right now inside the `performCasePurge` method for the Error/Warning cases)